infrastructure/policy/iam.tf

```diff
@@ -924,4 +924,67 @@ resource "aws_iam_role_policy" "authorize_lambda_invoke_policy" {
   ]
 }
 EOF
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# AWS IAM COGNITO TRIGGER PRE SIGN UP
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_iam_role" "verify_users_with_same_email_lambda_exec_role" {
+  name               = "${var.environment}_verify_users_with_same_email_exec_role"
+  path               = "/"
+  description        = "Allows Lambda Function to call AWS services on your behalf."
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "verify_users_with_same_email_policy" {
+  name        = "${var.environment}_pre_signup_trigger_policy"
+  description = "A policy to verify if it already exists email adress"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "VisualEditor0",
+      "Effect": "Allow",
+      "Action": [
+        "cognito-idp:ListUsers"
+      ],
+      "Resource": "${var.cognito_user_pool_arn}"
+      }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "verify_users_with_same_email_cognito_policy" {
+  role       = aws_iam_role.verify_users_with_same_email_lambda_exec_role.name
+  policy_arn = aws_iam_policy.verify_users_with_same_email_policy.arn
+}
+
+resource "aws_iam_role_policy_attachment" "verify_users_with_same_email_lambda_logs" {
+  role       = aws_iam_role.verify_users_with_same_email_lambda_exec_role.name
+  policy_arn = var.aws_iam_policy_logs_arn
+}
+
```

Comment:
The lambda'll need an aws_iam_role_policy to list users from Cognito, you can review authorizer lambda permissions.